### ISSUE TITLE
HampelFilter outlier detection in OptionalPassthrough

### ISSUE THREAD

**Issue by mateocomba123** [sktime/sktime#3683]
Hey! This is more a doubt than a bug.
I need to implement outlier detection and replacement on a pipeline, and found the OptionalPassthrough transformer that would work perfect, because I need precisely to try if the transformation works fine or not and choose the best (like an hyperparameter to tune, wether to use or not the transformation), so far so good. But, my concern is about how it does the transformation in the test set. Typically, the normal way to to things, in order to avoid training with the same data we test, is Initialize the transformer, train and transform with the train set (train_set = Transformer.fit_transform(train_set)) and then use that transformer already trained to transform the test data (test_set = Transformer.transform(test_set)). How does it works with HampelFilter, or StandardScaler, or other transformers that needs the same behaviour?
Another question regarding the OptionalPassthrough transformer is if it can be used on an "evaluate" function of sktime, or only can be implemented on ForecastingGridSearchCV. If so, it would be nice to have that on the userguide and I can help on the guide.
Thanks in advance!

**Comment by aiwalter** [comment]
hi @mateocomba123 , not sure if I got your question
1) `HampelFilter` has an empty `fit` method, but if you use it in a pipeline which is recommended then I dont see any issue?
2) You can just give the pipeline e.g. `TransformedTargetForecaster` to the `evaluate`

Not sure if that were your main questions? Otherwise pls separate a bit your text into separate questions. Thx

**Comment by mateocomba123** [comment]
Hey Martin, thanks so much for the quick response,
Regarding the HampelFilter, just to double check, if I use it in a pipeline
it means that when I fit the pipeline it will fit_transform the train data,
and when I use the predict attribute to forecast new values, it will
automatically do the transformation on that test, before making the
predictions/forecasts?
On the other hand, thanks for the heads up on giving the
TransformedTargetForecaster to the evaluate function! I will try that way

**Comment by aiwalter** [comment]
When you call `predict` there is no `transform` called in the pipeline but `inverse_transform` on model prediction. `HampelFilter` does not `inverse_transform` as it makes no sense to add outliers back again.

**Comment by mateocomba123** [comment]
Get it.
The doubt was thinking on the training/test splitting.
I want to make sure that the transformer is not "seeing" any test data in
advance. Meaning in the example of HampelFilter, that the transformer when
fitting doesnt "see" the values of the test set to calculate the standard
deviation and detect outliers, or, another similar example is the Standard
Scaler, that shouldn't use test observations to train the transformer, just
transform the test data with the information learnt from the training set.

**Comment by aiwalter** [comment]
Can we close this then?

**Comment by mateocomba123** [comment]
Yes, thanks a lot for the patience and help!If anything else I will open another issue :)El 31 oct. 2022, a la(s) 19:16, Martin Walter ***@***.***> escribió:﻿
Can we close this then?
